### PR TITLE
[air][cherry-pick] Disable head node syncing by default

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

Cherry-pick of https://github.com/ray-project/ray/pull/37142

See here for release test results: https://buildkite.com/ray-project/release-tests-pr/builds/45251#

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by bveeramani — APPROVED** [review]
LGTM. Will merge once Zhe approves

**Review by zhe-thoughts — APPROVED** [review]
Yes let's pick since this is a very critical fix. Thanks @justinvyu

**Comment by can-anyscale** [comment]
@justinvyu I think this breaks tune tests https://buildkite.com/ray-project/release-tests-branch/builds/1933

Logs:

```
Traceback (most recent call last):
  File "workloads/_tune_script.py", line 10, in <module>
    from ray.air.constants import REENABLE_DEPRECATED_SYNC_TO_HEAD_NODE
ImportError: cannot import name 'REENABLE_DEPRECATED_SYNC_TO_HEAD_NODE' from 'ray.air.constants' (/home/ray/anaconda3/lib/python3.8/site-packages/ray/air/constants.py)
```

**Pull request merged**
